sdk/core/core/inc/az_result.h

```diff
@@ -34,6 +34,9 @@ enum {
     } \
   } while (0)
 
+#define AZ_IGNORE_RESULT(exp) \
+  if (exp) {};
```

Comment:
It should be `do { … } while(0)`

Reply:
I don't think it will silence the GCC warning, but I can give it a try.
The reason I am introducing this macro is that on our current GCC version, and this is likely a bug, when you have AZ_NODISCARD. and  silence it with `(void)fn();`, it gives a warning.